urlmiddleware/middleware.py

```diff
@@ -42,10 +42,13 @@ def __call__(self, request):
     def process_view(self, request, view_func, view_args, view_kwargs):
         matched_middleware = self.get_matched_middleware(request.path)
         for middleware in matched_middleware:
+            middleware = import_string(middleware)
             if hasattr(middleware, 'process_view'):
-                response = middleware.process_view(request, view_func, view_args, view_kwargs)
+                instance = middleware(request)
+                response = instance.process_view(request, view_func, view_args, view_kwargs)
                 if response:
                     return response
+
         return None
 
     def process_exception(self, request, exception):
```

Comment:
@IhorKalhanov think you could add a similar fix to the process_exception function? I don't think we currently use it anywhere but makes sense to do it now while we're thinking about it

Reply:
Got it, I will do it